### DISCUSSION TITLE
请问服务配置文件无法启动解决办法

### DISCUSSION THREAD

**Discussion by njfds** [baidu/dperf#452] [Q&A] [answered]
server无法启动，大页都开启（512M大页开了128个），也绑定到了dpdk。请教专家是否知道解决的办法呢
![image](https://github.com/baidu/dperf/assets/170612252/7bddb88d-8c32-4e46-98fa-107cb91a1801)

dpdk-devbind.py --bind=vfio-pci 0000:7d:00.0
dpdk-devbind.py --bind=vfio-pci 0000:bd:00.0
Notice: 0000:7d:00.0 already bound to driver vfio-pci, skipping
Notice: 0000:bd:00.0 already bound to driver vfio-pci, skipping
[root@localhost dperf-main]# ./build/dperf -c server-nat.conf
EAL: Requested device 0000:bd:00.0 cannot be used
EAL: Bus (pci) probe failed.
not enough ports available: avail 0 require 1
port init fail
dpdk init fail
[root@localhost dperf-main]# cat server-nat.conf
mode            server
cpu             96-111
socket_mem      0,0,0,10240
duration        1d
#payload_size    1
protocol        http
#port   pci             addr            gateway(target ip)      target-mac
port    0000:bd:00.0    2.2.2.2         1.1.1.1                 8c:2a:8e:ff:62:90
#port   0000:81:00.0    2.2.2.2         10.0.0.1                8c:2a:8e:55:73:ff
client  1.1.1.1         1
server  2.2.2.2         16
listen  80              1

**Comment by pengjianzhang** [comment]
1. 用最新主干代码吗？
2. 配置log_level debug
3. 网卡与CPU是同一个NUMA吗？
4. 尽量用1G大页（除非系统不支持）
5. 是什么网卡？是否在DPDK支持列表中？
6. 如果喜欢dperf，请加个star，谢谢

**Comment by njfds** [comment]
1.dperf -v查询到是1.7版本
2.debug这个应该怎么配置呢？您是否有文档呢，我在代码没有找到log_level 这个文件
3.改成1个numa后还是不行
4.uos系统不支持1G
5.使用testpmd验证网卡是成功的。这个应该能说明dpdk支持该网卡吧
6.已经star，感谢专家

**Reply by pengjianzhang** [reply]
配置文件里写一行: log_level debug
https://dperf.org/doc/html/configuration#section41

**Reply by njfds** [reply]
![1](https://github.com/baidu/dperf/assets/170612252/a0e98a96-b95d-45f2-918a-09d85351f284)
您好加入了后增加了一些打印信息，您能看出来问题的所在吗
 ./build/dperf -c server-nat.conf
EAL: Detected 128 lcore(s)
EAL: Detected 4 NUMA nodes
EAL: Detected static linkage of DPDK
EAL: Multi-process socket /var/run/dpdk/5524/mp_socket
EAL: Selected IOVA mode 'VA'
EAL: No available hugepages reported in hugepages-2048kB
EAL: Probing VFIO support...
EAL: VFIO support initialized
EAL:   using IOMMU type 1 (Type 1)
EAL: Ignore mapping IO port bar(1)
EAL: Ignore mapping IO port bar(3)
EAL: Probe PCI driver: net_hns3 (19e5:a222) device: 0000:bd:00.0 (socket 2)
EAL: Releasing pci mapped resource for 0000:bd:00.0
EAL: Calling pci_unmap_resource for 0000:bd:00.0 at 0x920000000
EAL: Calling pci_unmap_resource for 0000:bd:00.0 at 0x920010000
EAL: Requested device 0000:bd:00.0 cannot be used
EAL: Bus (pci) probe failed.
not enough ports available: avail 0 require 1
port init fail
dpdk init fail

**Comment by pengjianzhang** [comment]
估计是cpu用的不对，只配一个cpu，尝试用各个numa上cpu 都试一遍

**Reply by njfds** [reply, marked as answer]
已解决，注释掉socker_mem这一行就可以了

**Reply by pengjianzhang** [reply]
👍🏻 have fun
请向周围小伙伴宣传一下 谢谢
